### PR TITLE
CVSB-11033 - Modified retro reports to accomodate coif and specialist tests

### PR DESCRIPTION
**Pull request by oalexandru**

### CODE REVIEW

File: src/services/RetroGenerationService.ts [hunk 1]
```diff
@@ -79,49 +79,61 @@ class RetroGenerationService {
...
-                    + "Additional test type notes: " + additionalTestTypeNotes + ";\r\n"
-                    + (testType.additionalNotesRecorded ? (testType.additionalNotesRecorded + ";") : "");
-                }
-                if (event.activityType === ActivityType.TIME_NOT_TESTING) {
-                  // Populate wait activities in the report
-                  const detailsTemplate: any = template.reportTemplate.activityDetails[i];
-                  const waitActivityResult: any = event.activity;
-                  let waitReasons: string = "";
-                  let additionalNotes: string = "";
+                                    defectsDetails = defectsDetails + " " + testType.defects[key].deficiencyRef + " (" +
```
Comment (m-ujaffer) [inline src/services/RetroGenerationService.ts]: indentation issue

File: src/services/RetroGenerationService.ts [hunk 2]
```diff
@@ -365,9 +377,14 @@ class RetroGenerationService {
    * @param testType
    */
   private isPassingLECTestType(testType: any): boolean {
-    const lecTestTypeIds = ["39", "44", "45"];
-    return lecTestTypeIds.includes(testType.testTypeId) && testType.testResult === TEST_RESULT_STATES.PASS;
-  }
+        const lecTestTypeIds = ["39", "44", "45"];
+        return lecTestTypeIds.includes(testType.testTypeId) && testType.testResult === TEST_RESULT_STATES.PASS;
+    }
+
```
Comment (m-ujaffer) [inline src/services/RetroGenerationService.ts]: method docs missing

File: src/services/RetroGenerationService.ts [hunk 3]
```diff
@@ -79,49 +79,61 @@ class RetroGenerationService {
...
+                                    }
+                                if (testType.additionalCommentsForAbandon) {
+                                        additionalCommentsAbandon = `Additional comments for abandon: ${testType.additionalCommentsForAbandon};\r\n`;
+                                    }
+                                if (this.isPassingLECTestType(testType)) {
+                                  LECNotes = "Modification type: " + testType.modType.code.toUpperCase() + "\r\n"
+                                    + "Fuel type: " + testType.fuelType + "\r\n"
+                                    + "Emission standards: " + testType.emissionStandard + "\r\n";
+                                }
+                                let customDefects = "";
```
Comment (m-ujaffer) [inline src/services/RetroGenerationService.ts]: suggest to rename as  customDefectsStr.

File: src/services/RetroGenerationService.ts [hunk 4]
```diff
@@ -79,49 +79,61 @@ class RetroGenerationService {
...
+                                detailsTemplate.expiryDate.value = testType.testExpiryDate ? moment(testType.testExpiryDate).tz(TimeZone.LONDON).format("DD/MM/YYYY") : "";
+                                detailsTemplate.preparerId.value = testResult.preparerId;
+                                detailsTemplate.failureAdvisoryItemsQAIComments.value = defects
+                                                                                      + reasonForAbandoning
+                                                                                      + additionalCommentsAbandon
+                                                                                      + LECNotes
+                                                                                      + "Additional test type notes: " + additionalTestTypeNotes + ";\r\n"
+                                                                                      + (testType.additionalNotesRecorded ? (testType.additionalNotesRecorded + ";") : "")
+                                                                                      + customDefects ? "\r\nCustom defects:\r\n" + customDefects : "";
```
Comment (m-ujaffer) [inline src/services/RetroGenerationService.ts]: move `"\r\nCustom defects:\r\n" + customDefects` inside the `if (testType.customDefects)` to remove conditional check over here.

File: src/services/RetroGenerationService.ts (hunk 2 above)
Comment (oalexandru) [reply src/services/RetroGenerationService.ts]: I believe you where referring to isTestTypeCoifWithAnnualTestOrCoifWithAnnualTestRetest() method. fixed

File: src/services/RetroGenerationService.ts (hunk 1 above)
Comment (oalexandru) [reply src/services/RetroGenerationService.ts]: fixed

File: src/services/RetroGenerationService.ts (hunk 4 above)
Comment (oalexandru) [reply src/services/RetroGenerationService.ts]: fixed

File: src/services/RetroGenerationService.ts (hunk 3 above)
Comment (oalexandru) [reply src/services/RetroGenerationService.ts]: done

File: src/services/RetroGenerationService.ts [hunk 5]
```diff
@@ -106,30 +117,29 @@ class RetroGenerationService {
...
-                    + additionalCommentsAbandon
-                    + LECNotes
-                    + "Additional test type notes: " + additionalTestTypeNotes + ";\r\n"
-                    + (testType.additionalNotesRecorded ? (testType.additionalNotesRecorded + ";") : "");
+                                                                                      + reasonForAbandoning
+                                                                                      + additionalCommentsAbandon
+                                                                                      + LECNotes
+                                                                                      + "Additional test type notes: " + additionalTestTypeNotes + ";\r\n"
+                                                                                      + (testType.additionalNotesRecorded ? (testType.additionalNotesRecorded + ";") : "")
+                                                                                      + customDefectsStr;
```
Comment (KHodderDVSA) [inline src/services/RetroGenerationService.ts]: Why the huge indentation?

Comment (oalexandru) [reply src/services/RetroGenerationService.ts]: auto-fix indentation is messing with me. fixed

**Pull request closed without merging**